### PR TITLE
Allow only Japanese phone numbers for Japanese accounts

### LINKED ISSUE #6911: Settings cannot be saved when non-Japanese phone is used for Japanese accounts
### Describe the bug
A Japanese phone number is required for Japanese accounts. Usage of non-Japanese phone numbers on the Settings page results in a successful message, but the settings are not saved.

### To Reproduce
1. Connect a Japanese account.
2. Navigate to `Payments` -> `Settings`.
3. Enter a non-Japanese phone number on the `Support phone` field.
4. Change a random setting on the page.
5. Save the settings and verify a successful message is displayed.
6. Reload the page and verify the phone and the setting are not updated.

### Expected behavior
Saving the changes with a non-Japanese phone number should trigger an error.

### PR DESCRIPTION
**Pull request by mordeth**
Fixes #6911

#### Changes proposed in this Pull Request
An error message is displayed when a non-Japanese phone is used for Japanese accounts in the `Support phone` field on the `Settings` page.

Screenshot 2023-08-03 at 14 50 07

Screenshot 2023-08-03 at 14 50 28

#### Testing instructions
- Use a Japanese account.
- Navigate to `Payments` -> `Settings`.
- Verify only Japan is available on the `Support phone` input field.
- Enter a non-Japanese phone, e.g. '+359878111111'.
- Verify an error is triggered on Save settings.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6912` or branch name `fix/japan-phone-validation` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/japan-phone-validation) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: efb34f9fcd5ecbf466168bec3e720f166df74057
- Build time: 2023-08-04 11:32:53 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/admin/class-wc-rest-payments-settings-controller.php [hunk 1]
```diff
@@ -335,6 +349,16 @@ public function validate_business_support_phone( string $value, WP_REST_Request
 			);
 		}
 
+		// Japan accounts require Japanese phone numbers.
+		if ( 'JP' === $this->account->get_account_country() ) {
```
Comment (dpaun1985) [inline includes/admin/class-wc-rest-payments-settings-controller.php:353]: we can have only 1 if here

**Review by dpaun1985 — APPROVED** [review]
✅ LGTM. I left a minor comment.

File: client/settings/phone-input/index.tsx [hunk 2]
```diff
@@ -70,14 +70,29 @@ const PhoneNumberInput = ( {
 			}
 		};
 
+		let onlyCountries = {
```
Comment (elazzabi) [inline client/settings/phone-input/index.tsx]: I think the name of the variable is a little misleading. Its name is `onlyCountries` and inside it there is an object with `onlyCountries`... What do you think about naming this variable `countries` or `phoneCountries` or something like this?

File: client/settings/phone-input/index.tsx [hunk 3]
```diff
@@ -70,14 +70,29 @@ const PhoneNumberInput = ( {
...
+			initialCountry: 'US',
+			onlyCountries: [],
+		};
+
+		const accountCountry = wcpaySettings?.accountStatus?.country ?? '';
+		if ( accountCountry === 'JP' ) {
```
Comment (elazzabi) [inline client/settings/phone-input/index.tsx:81]:
```suggestion
		// Special case for Japan: Only Japanese phone numbers are accepted by Stripe 
		if ( accountCountry === 'JP' ) {
```

**Review by elazzabi — APPROVED** [review]
I left a few comments. Once fixed we should be good to go.

Pre-approving to unblock things.

**Review by anu-rock — APPROVED** [review]
LGTM 🎉 

___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | ✅ All good |
| Naming  | 👍 Almost good (I agree with @elazzabi's comment about renaming `onlyCountries`) |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Testing instructions  | ✅ All good |
| Manual testing  | ✅ Works as expected |
| Edge cases  | ✅ All good |

#### Scenarios tested

| Scenario  | Result |
| ------------- | ------------- |
| +81 is the only option in dropdown and is selected by default  | ✅ |
| All [valid phone numbers](https://www.random-name-generator.com/japanese-phone-number-generator) work  | ✅ |
| Adding or removing a digit triggers an error message  | ✅ |
| Force entering another country's number (eg. +919012345678) and clicking Save button triggers an error message  | ✅ |

**Comment by rafaelzaleski** [comment]
@mordeth, this commit is breaking my local dev environment. This is from the checkout page when WooPay is enabled. The `PhoneNumberInput` component is used there.

```
react-dom.js?ver=18.2.0:22849 Uncaught ReferenceError: wcpaySettings is not defined
    at eval (index.tsx:78:26)
    at commitHookEffectListMount (react-dom.js?ver=18.2.0:23160:28)
    at commitPassiveMountOnFiber (react-dom.js?ver=18.2.0:24936:15)
    at commitPassiveMountEffects_complete (react-dom.js?ver=18.2.0:24901:11)
    at commitPassiveMountEffects_begin (react-dom.js?ver=18.2.0:24888:9)
    at commitPassiveMountEffects (react-dom.js?ver=18.2.0:24876:5)
    at flushPassiveEffectsImpl (react-dom.js?ver=18.2.0:27049:5)
    at flushPassiveEffects (react-dom.js?ver=18.2.0:26994:16)
    at flushSync (react-dom.js?ver=18.2.0:26186:7)
    at finishEventHandler (react-dom.js?ver=18.2.0:3963:7)
eval @ index.tsx:78
commitHookEffectListMount @ react-dom.js?ver=18.2.0:23160
commitPassiveMountOnFiber @ react-dom.js?ver=18.2.0:24936
commitPassiveMountEffects_complete @ react-dom.js?ver=18.2.0:24901
commitPassiveMountEffects_begin @ react-dom.js?ver=18.2.0:24888
commitPassiveMountEffects @ react-dom.js?ver=18.2.0:24876
flushPassiveEffectsImpl @ react-dom.js?ver=18.2.0:27049
flushPassiveEffects @ react-dom.js?ver=18.2.0:26994
flushSync @ react-dom.js?ver=18.2.0:26186
finishEventHandler @ react-dom.js?ver=18.2.0:3963
batchedUpdates @ react-dom.js?ver=18.2.0:3981
dispatchEventForPluginEventSystem @ react-dom.js?ver=18.2.0:9297
dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay @ react-dom.js?ver=18.2.0:6475
dispatchEvent @ react-dom.js?ver=18.2.0:6467
dispatchDiscreteEvent @ react-dom.js?ver=18.2.0:6440
react-dom.js?ver=18.2.0:18697 The above error occurred in the <PhoneNumberInput> component:

    at PhoneNumberInput (webpack-internal:///./client/settings/phone-input/index.tsx:44:26)
    at div
    at div
    at Container (webpack-internal:///./client/components/woopay/save-user/container.js:16:5)
    at CheckoutPageSaveUser (webpack-internal:///./client/components/woopay/save-user/checkout-page-save-user.js:58:5)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
logCapturedError @ react-dom.js?ver=18.2.0:18697
update.callback @ react-dom.js?ver=18.2.0:18730
callCallback @ react-dom.js?ver=18.2.0:13933
commitUpdateQueue @ react-dom.js?ver=18.2.0:13954
commitLayoutEffectOnFiber @ react-dom.js?ver=18.2.0:23401
commitLayoutMountEffects_complete @ react-dom.js?ver=18.2.0:24698
commitLayoutEffects_begin @ react-dom.js?ver=18.2.0:24684
commitLayoutEffects @ react-dom.js?ver=18.2.0:24622
commitRootImpl @ react-dom.js?ver=18.2.0:26833
commitRoot @ react-dom.js?ver=18.2.0:26692
performSyncWorkOnRoot @ react-dom.js?ver=18.2.0:26127
flushSyncCallbacks @ react-dom.js?ver=18.2.0:12052
flushPassiveEffectsImpl @ react-dom.js?ver=18.2.0:27070
flushPassiveEffects @ react-dom.js?ver=18.2.0:26994
flushSync @ react-dom.js?ver=18.2.0:26186
finishEventHandler @ react-dom.js?ver=18.2.0:3963
batchedUpdates @ react-dom.js?ver=18.2.0:3981
dispatchEventForPluginEventSystem @ react-dom.js?ver=18.2.0:9297
dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay @ react-dom.js?ver=18.2.0:6475
dispatchEvent @ react-dom.js?ver=18.2.0:6467
dispatchDiscreteEvent @ react-dom.js?ver=18.2.0:6440
react-dom.js?ver=18.2.0:12066 Uncaught ReferenceError: wcpaySettings is not defined
    at eval (index.tsx:78:26)
    at commitHookEffectListMount (react-dom.js?ver=18.2.0:23160:28)
    at commitPassiveMountOnFiber (react-dom.js?ver=18.2.0:24936:15)
    at commitPassiveMountEffects_complete (react-dom.js?ver=18.2.0:24901:11)
    at commitPassiveMountEffects_begin (react-dom.js?ver=18.2.0:24888:9)
    at commitPassiveMountEffects (react-dom.js?ver=18.2.0:24876:5)
    at flushPassiveEffectsImpl (react-dom.js?ver=18.2.0:27049:5)
    at flushPassiveEffects (react-dom.js?ver=18.2.0:26994:16)
    at flushSync (react-dom.js?ver=18.2.0:26186:7)
    at finishEventHandler (react-dom.js?ver=18.2.0:3963:7)
```

**Comment by dpaun1985** [comment]
> @mordeth, this commit is breaking my local dev environment. This is from the checkout page when WooPay is enabled. The `PhoneNumberInput` component is used there.
> 
> ```
> react-dom.js?ver=18.2.0:22849 Uncaught ReferenceError: wcpaySettings is not defined
>     at eval (index.tsx:78:26)
>     at commitHookEffectListMount (react-dom.js?ver=18.2.0:23160:28)
>     at commitPassiveMountOnFiber (react-dom.js?ver=18.2.0:24936:15)
>     at commitPassiveMountEffects_complete (react-dom.js?ver=18.2.0:24901:11)
>     at commitPassiveMountEffects_begin (react-dom.js?ver=18.2.0:24888:9)
>     at commitPassiveMountEffects (react-dom.js?ver=18.2.0:24876:5)
>     at flushPassiveEffectsImpl (react-dom.js?ver=18.2.0:27049:5)
>     at flushPassiveEffects (react-dom.js?ver=18.2.0:26994:16)
>     at flushSync (react-dom.js?ver=18.2.0:26186:7)
>     at finishEventHandler (react-dom.js?ver=18.2.0:3963:7)
> eval @ index.tsx:78
> commitHookEffectListMount @ react-dom.js?ver=18.2.0:23160
> commitPassiveMountOnFiber @ react-dom.js?ver=18.2.0:24936
> commitPassiveMountEffects_complete @ react-dom.js?ver=18.2.0:24901
> commitPassiveMountEffects_begin @ react-dom.js?ver=18.2.0:24888
> commitPassiveMountEffects @ react-dom.js?ver=18.2.0:24876
> flushPassiveEffectsImpl @ react-dom.js?ver=18.2.0:27049
> flushPassiveEffects @ react-dom.js?ver=18.2.0:26994
> flushSync @ react-dom.js?ver=18.2.0:26186
> finishEventHandler @ react-dom.js?ver=18.2.0:3963
> batchedUpdates @ react-dom.js?ver=18.2.0:3981
> dispatchEventForPluginEventSystem @ react-dom.js?ver=18.2.0:9297
> dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay @ react-dom.js?ver=18.2.0:6475
> dispatchEvent @ react-dom.js?ver=18.2.0:6467
> dispatchDiscreteEvent @ react-dom.js?ver=18.2.0:6440
> react-dom.js?ver=18.2.0:18697 The above error occurred in the <PhoneNumberInput> component:
> 
>     at PhoneNumberInput (webpack-internal:///./client/settings/phone-input/index.tsx:44:26)
>     at div
>     at div
>     at Container (webpack-internal:///./client/components/woopay/save-user/container.js:16:5)
>     at CheckoutPageSaveUser (webpack-internal:///./client/components/woopay/save-user/checkout-page-save-user.js:58:5)
> 
> Consider adding an error boundary to your tree to customize error handling behavior.
> Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
> logCapturedError @ react-dom.js?ver=18.2.0:18697
> update.callback @ react-dom.js?ver=18.2.0:18730
> callCallback @ react-dom.js?ver=18.2.0:13933
> commitUpdateQueue @ react-dom.js?ver=18.2.0:13954
> commitLayoutEffectOnFiber @ react-dom.js?ver=18.2.0:23401
> commitLayoutMountEffects_complete @ react-dom.js?ver=18.2.0:24698
> commitLayoutEffects_begin @ react-dom.js?ver=18.2.0:24684
> commitLayoutEffects @ react-dom.js?ver=18.2.0:24622
> commitRootImpl @ react-dom.js?ver=18.2.0:26833
> commitRoot @ react-dom.js?ver=18.2.0:26692
> performSyncWorkOnRoot @ react-dom.js?ver=18.2.0:26127
> flushSyncCallbacks @ react-dom.js?ver=18.2.0:12052
> flushPassiveEffectsImpl @ react-dom.js?ver=18.2.0:27070
> flushPassiveEffects @ react-dom.js?ver=18.2.0:26994
> flushSync @ react-dom.js?ver=18.2.0:26186
> finishEventHandler @ react-dom.js?ver=18.2.0:3963
> batchedUpdates @ react-dom.js?ver=18.2.0:3981
> dispatchEventForPluginEventSystem @ react-dom.js?ver=18.2.0:9297
> dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay @ react-dom.js?ver=18.2.0:6475
> dispatchEvent @ react-dom.js?ver=18.2.0:6467
> dispatchDiscreteEvent @ react-dom.js?ver=18.2.0:6440
> react-dom.js?ver=18.2.0:12066 Uncaught ReferenceError: wcpaySettings is not defined
>     at eval (index.tsx:78:26)
>     at commitHookEffectListMount (react-dom.js?ver=18.2.0:23160:28)
>     at commitPassiveMountOnFiber (react-dom.js?ver=18.2.0:24936:15)
>     at commitPassiveMountEffects_complete (react-dom.js?ver=18.2.0:24901:11)
>     at commitPassiveMountEffects_begin (react-dom.js?ver=18.2.0:24888:9)
>     at commitPassiveMountEffects (react-dom.js?ver=18.2.0:24876:5)
>     at flushPassiveEffectsImpl (react-dom.js?ver=18.2.0:27049:5)
>     at flushPassiveEffects (react-dom.js?ver=18.2.0:26994:16)
>     at flushSync (react-dom.js?ver=18.2.0:26186:7)
>     at finishEventHandler (react-dom.js?ver=18.2.0:3963:7)
> ```

Thank you for reporting this @rafaelzaleski . I created https://github.com/Automattic/woocommerce-payments/issues/6952 and we will fix it ASAP.

**Pull request merged**